### ISSUE TITLE
False positive: `UnnecessaryAssertionInspection` on `assert::instanceOf(SomeSubType::class, $some->typedCall())`

### ISSUE THREAD

**Issue by Ocramius** [kalessil/phpinspectionsea#1048]
First, thank you for reporting a bug or making a request. That takes time and we appreciate that!

Let's start with common details:
 
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive                          |
| Plugin         | Php Inspections (EA Ultimate), 2.0.8 |
| Language level | 7.2                        |

Consider following code:

```php
interface T {}
class A implements T {}
class B implements T {}
class C implements T {}
function foo() : T {
    $types = [new A(), new B(), new C()];
    return $types[array_rand($types)];
}
```

This unit test will report an `UnnecessaryAssertionInspection`:

```
public function FooTest extends TestCase
{
    public function testFoo() : void
    {
        self::assertInstanceOf(A::class, foo());
    }
}
```

**Comment by kalessil** [comment]
Good catch Marco, I'll take care of fixing this.

**Comment by kalessil** [comment]
Fixed!
